FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -116,181 +123,160 @@ + (instancetype)appDistribution {
 
   // In the component creation block, we return an instance of `FIRAppDistribution`. Cast it and
   // return it.
+  FIRFADDebugLog(@"Instance returned: %@", instance);
   return (FIRAppDistribution *)instance;
 }
 
 - (void)signInTesterWithCompletion:(void (^)(NSError *_Nullable error))completion {
-  NSURL *issuer = [NSURL URLWithString:kIssuerURL];
-
-  [OIDAuthorizationService
-      discoverServiceConfigurationForIssuer:issuer
-                                 completion:^(OIDServiceConfiguration *_Nullable configuration,
-                                              NSError *_Nullable error) {
-                                   [self handleOauthDiscoveryCompletion:configuration
-                                                                  error:error
-                                        appDistributionSignInCompletion:completion];
-                                 }];
-}
+  FIRFADDebugLog(@"Prompting tester for sign in");
 
-- (void)signOutTester {
-  NSError *error;
-  BOOL didClearAuthState = [FIRAppDistributionAuthPersistence clearAuthState:&error];
-  // TODO (schnecle): Add in FIRLogger to report when we have failed to clear auth state
-  if (!didClearAuthState) {
-    NSLog(@"Error clearing token from keychain: %@", [error localizedDescription]);
+  if ([self isTesterSignedIn]) {
+    completion(nil);
+    return;
   }
 
-  self.authState = nil;
-  self.isTesterSignedIn = false;
-}
+  [self.UIService initializeUIState];
+  FIRInstallations *installations = [FIRInstallations installations];
```

Comment:
As far as I can see `FIRAppDistribution` supports multiple `FIRApps`. If so, then [installationsWithApp:](https://github.com/firebase/firebase-ios-sdk/blob/0ed26aedb85f3f3fd37b0077133347183db5de7a/FirebaseInstallations/Source/Library/Public/FIRInstallations.h#L74) method should be used to get an instance for the app.

Reply:
At the moment, I don't believe we support multiple apps: https://github.com/firebase/firebase-ios-sdk/blob/619864e4cb965c0ac07163c7dd6cf080df6f1b48/FirebaseAppDistribution/Sources/FIRAppDistribution.m#L92

Is there somewhere else that indicates that we do support multiple apps?

Reply:
Ah, I see, sorry, I was confused by the presence of the [initializer accepting an app](https://github.com/firebase/firebase-ios-sdk/blob/619864e4cb965c0ac07163c7dd6cf080df6f1b48/FirebaseAppDistribution/Sources/FIRAppDistribution.m#L68).

Reply:
No worries - @samedson  had some good feedback on readability here, and I think we can make this a lot clearer with a little refactoring, but we'll do that in a subsequent PR if that's ok as it will be another large one.

Reply:
SGTM